docs/DeveloperGuide.md

```diff
@@ -718,29 +703,59 @@ Use case ends.
 * 2a. System detects an invalid index entered.
   * 2a1. System shows an error message.
   * 2a2. User enters a new delete request. <br>
-  Steps 2a1- 2a2 are repeated until the data entered are correct.<br>
+  Steps 2a1- 2a2 are repeated until the data entered is correct.<br>
   Use case resumes from step 3. <br>
 
-**Use Case UC7: Add emergency contact**
+**Use Case UC6 - Add note**
+
+**MSS**
+1. User <ins>lists all contacts (UC2).</ins>
+2. User enters an index to add a note associated with a contact
+3. System adds the note to the contact.<br>
+Use case ends.
+
+**Extensions**
+* 2a. System detects an invalid index entered.
+  * 2a1. System shows an error message.
+  * 2a2. User tries to add a note to a contact again.<br>
+  Steps 2a1- 2a2 are repeated until the data entered is correct. <br>
+  Use case resumes from step 3.
+
+
+**Use Case UC7 - Record payment**
 
 **MSS**
-1. User <ins>lists all contacts (UC3).</ins>
-2. User adds a contact to the emergency contact list.
-3. System adds the contact into the emergency contact list. <br>
+1. User <ins>lists all contacts (UC2).</ins>
+2. User enters an index to record an amount of money paid to a contact.
```

Comment:
afaik, it is not necessarily `paid` but also, there is a concept of `owe`. perhaps we can combine them into this single use case?

```suggestion
2. User enters an index of the specified contact alongside with the amount of money paid/owed to that contact.
```